core/new-gui/src/app/dashboard/component/feature-container/user-project-section/project-page/project-page.component.ts

```diff
@@ -0,0 +1,306 @@
+  }
+
+  /**
+   * duplicate the current workflow. A new record will appear in frontend
+   * workflow list and backend database.
+   * 
+   * Modified to also add new workflow to project
+   */
+
+  public onClickDuplicateWorkflow({ workflow: { wid } }: DashboardWorkflowEntry): void {
+    if (wid) {
+      this.workflowPersistService
+        .duplicateWorkflow(wid)
+        .pipe(
+          concatMap((duplicatedWorkflowInfo: DashboardWorkflowEntry) => {
+            this.workflows.push(duplicatedWorkflowInfo); 
+            return this.userProjectService.addWorkflowToProject(this.pid, duplicatedWorkflowInfo.workflow.wid!);
+          }),
+          catchError((err : unknown) => {
+            throw err;
+          }),
+          untilDestroyed(this))
+        .subscribe(
+          () => {},
+          // @ts-ignore // TODO: fix this with notification component
+          (err: unknown) => alert(err.error)
+        );
+    }
+  }
+
+  /**
+   * openNgbdModalDeleteWorkflowComponent trigger the delete workflow
+   * component. If user confirms the deletion, the method sends
+   * message to frontend and delete the workflow on frontend. It
+   * calls the deleteProject method in service which implements backend API.
+   */
+   public onClickDeleteWorkflow({ workflow }: DashboardWorkflowEntry): void {
+    const modalRef = this.modalService.open(NgbdModalDeleteWorkflowComponent);
+    modalRef.componentInstance.workflow = cloneDeep(workflow);
+
+    from(modalRef.result)
+      .pipe(untilDestroyed(this))
+      .subscribe((confirmToDelete: boolean) => {
+        const wid = workflow.wid;
+        if (confirmToDelete && wid !== undefined) {
+          this.workflowPersistService
+            .deleteWorkflow(wid)
+            .pipe(untilDestroyed(this))
+            .subscribe(
+              _ => {
+                this.workflows = this.workflows.filter(
+                  workflowEntry => workflowEntry.workflow.wid !== wid
+                );
+              },
+              // @ts-ignore // TODO: fix this with notification component
+              (err: unknown) => alert(err.error)
+            );
+        }
+      });
+  }
+
+
+
+  // ----------------- for file card
+  public addFileSizeUnit(fileSize: number): string {
+    return this.userFileService.addFileSizeUnit(fileSize);
+  }
+
+  public confirmEditFileName(
+    dashboardUserFileEntry: DashboardUserFileEntry,
+    name: string,
+    index: number
+  ): void {
+    const {
+      file: { fid },
+    } = dashboardUserFileEntry;
+    this.userFileService
+      .updateFileName(fid, name)
+      .pipe(untilDestroyed(this))
+      .subscribe(
+        () => {
+          this.userFileService.refreshDashboardUserFileEntries();
+          this.userProjectService.refreshFilesOfProject(this.pid); // -- perform appropriate call for project page
+        },
+        (err: unknown) => {
+          // @ts-ignore // TODO: fix this with notification component
```

Comment:
is this TODO still needed?

Reply:
removed TODO and used notificationService